### DISCUSSION TITLE
Ignore windows from other displays for layout rules

### DISCUSSION THREAD

**Discussion by jschuur** [rxhanson/RectanglePro-Community#533] [Ideas]
I have an app layout for Chrome/iTerm/VS Code for my main monitor, but when I trigger it, I don't want it to move other Chrome  windows on different displays to the current display. I only want those in the current display/space to move into the designated layout spots. Other Chrome windows on other displays should stay where they are.

It seems smart enough to not move windows in other spaces, but it still moves those from another display.

Using Rectangle Pro 3.0.30 (171).

**Comment by rxhanson** [comment]
That's a fair point. I'll think through this one for the next release.

**Reply by jschuur** [reply]
> That's a fair point. I'll think through this one for the next release.

Nice! If it helps, when I went to see if this was supported, I was looking for a checkbox here that said something like 'Only use windows current display/space'

![CleanShot 2024-07-24 at 11 31 35](https://github.com/user-attachments/assets/ea671fbf-a3dc-4be0-950a-788757d607ac)

**Comment by jschuur** [comment]
Maybe an alternate (or parallel, since it's subtly different)  way to handle this here would be to introduce a per layout entry option called 'Only execute on frontmost(?) window'.

Supposed I had several VS Code/iTerm/Chrome windows open from each of those apps, and I wanted to arrange a subset of them (one per app) into a layout on a display. If you could identify the latest window from each of those apps to apply a layout too, I would avoid roping in other windows and moving/resizing those.

Right now, the 'Only execute on frontmost app' option applies to the whole layout, so this isn't useful here.

This is a real world example, and was actually one of the first things I looked for when I tried out layouts. If I am switching between a few projects and have a set of those three apps open for each of them, I'd like to give each project its own space with an VS Code/iTerm/Chrome window for each of them, arranged in the same layout.

**Reply by rxhanson** [reply]
Thanks for the additional thoughts here. The only catch with having it take into account "frontmost window" would be that there could be a situation where that frontmost or last window is on a different display. It sounds like keeping certain windows on their current display is really what you're after?

**Reply by jschuur** [reply]
> Thanks for the additional thoughts here. The only catch with having it take into account "frontmost window" would be that there could be a situation where that frontmost or last window is on a different display. It sounds like keeping certain windows on their current display is really what you're after?

Good point. In this case, the most useful would be 'apply this to all windows in the current display AND space'.

It could be a dropdown that says 'Apply this' with options for 'foremost app, all on current display and all on current space', depending on how you feel about the additional cognitive overload.
